spec/requests/profile_spec.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+describe "Profile", type: :request do
+  describe "#index" do
+    context "when user is not logged in" do
+      before { get profile_index_path, params: {uid: "user-id"} }
+      it { expect(response).to redirect_to(new_user_session_path) }
+    end
+
+    context "when user is logged in" do
+      let(:user1) { create(:user1) }
+      before { sign_in user1 }
+      it "renders index template" do
+        get profile_index_path, params: {uid: user1.uid}
+        expect(response).to render_template(:index)
+      end
+
+      context "when profile belongs to logged in user" do
+        it "assigns profile instance to be that of logged in user" do
+          get profile_index_path, params: {uid: user1.uid}
+          expect(assigns(:profile)).to eq(user1)
+        end
+
+        context "when user has no moments and strategies" do
+          it "assigns stories instance with an empty array" do
+            get profile_index_path, params: {uid: user1.uid}
+            expect(assigns(:stories)).to match_array([])
+          end
+        end
+
+        context "when user has moments or strategies" do
+          it "assigns stories instance with user moments or strategies" do
+            stories = [build_stubbed(:moment), build_stubbed(:strategy)]
+            allow_any_instance_of(ProfileController)
+              .to receive(:get_stories).and_return(stories)
+            get profile_index_path, params: {uid: user1.uid}
+            expect(assigns(:stories)).to match_array(stories)
```

Comment:
I'd probably change `assigns` to response body and make sure that it contains a name or something of each story since `assigns` is dependent on `rails-controller-testing` gem.

Reply:
I'm going to change this test to test that the `response.body` has the Stories section and the generated moment and strategy. The view template will only render that section if there are stories.